pkg/store/proxy.go

```diff
@@ -130,7 +140,12 @@ func (s *ProxyStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesSe
 		}
 	}
 
-	return g.Wait()
+	if err := g.Wait(); err != nil {
```

Comment:
Is it possible that your `g.Go(...)` function would terminate before `range respCh` is executed? This would mean that `respCh` would get closed before being iterated and `range respCh` wouldn't do anything?

Reply:
> This would mean that respCh would get closed before being iterated and range respCh wouldn't do anything?
Yes. And also this is the desired behavior.

If you write X messages to channel and close it channel and after that you will read from channel, you will receive X messages first then you will be notified that the channel is closed.

See this: https://dave.cheney.net/2014/03/19/channel-axioms
```
Once a channel is closed, and all values drained from its buffer, the channel will always return zero values immediately.
```